2D3D_UnityProject/Assets/Scripts/Puzzles/Reflection/ReflectableLightHead.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ReflectableLightHead : MonoBehaviour
+{
+    //We need these two objects for this script to work as intended, but no other script needs to use them, thus we'll make them private.
+    [SerializeField]
+    private GameObject toGenerateOnReflect;
+    [SerializeField]
+    private GameObject myBeam;
+
+    //We need the immediate child to be public for light nodes to work properly, but we absolutely do NOT want anyone setting it before running the game. It's intended function is to keep track of reflections created, which would be completely screwed up by setting this in the editor.
+    //For this reason, we make it nonserialized, thus making such a mistake impossible.
+    [System.NonSerialized]
+    public GameObject immediateChild = null;
+
+    //These are needed in order to calculate reflections and rays
+    RaycastHit rcHit;
+    Vector3 HitPosit = Vector3.zero;
```

Comment:
The standard abbreviation for the word position is 'pos', not 'posit', should probably be replaced for clarity.